src/lib/ext-activitylog.js

```diff
@@ -3,26 +3,199 @@ import { save } from './save-load.js';
+            this.apiTypeCheckboxList
+          );
+          break;
+        default:
+          throw new Error(
+            `wrong event target - ${event.currentTarget.tagName}`
+          );
+      }
+    } else if (event.type === 'change') {
+      this.handleChangeOnCheckbox(event);
+    } else {
+      throw new Error(`wrong event type - ${event.type}`);
+    }
+  }
+
+  toggleFilterListDisplay(toggleBtn, checkboxList) {
+    checkboxList.hidden = !checkboxList.hidden;
+    this.toggleFilterBtnIcon(toggleBtn, checkboxList.hidden);
+  }
+
+  toggleFilterBtnIcon(element, isListHidden) {
+    isListHidden
+      ? element
+          .querySelector('.arrow-down')
+          .classList.replace('arrow-down', 'arrow-right')
+      : element
+          .querySelector('.arrow-right')
+          .classList.replace('arrow-right', 'arrow-down');
+  }
+
+  handleChangeOnCheckbox(event) {
+    let target = null;
+    switch (event.currentTarget) {
+      case this.extCheckboxList:
+        target = 'id';
+        break;
+      case this.viewTypeCheckboxList:
+        target = 'viewType';
+        break;
+      case this.apiTypeCheckboxList:
+        target = 'type';
+        break;
+      default:
+        break;
+    }
+
+    if (target) {
+      const filterObj = {
+        filterParam: { value: event.target.closest('input').value, target },
+        checked: event.target.closest('input').checked,
+      };
+
+      const filterEvent = new CustomEvent('filter', { detail: filterObj });
+      event.currentTarget.dispatchEvent(filterEvent);
+    }
+  }
+
+  handleNewLog(object) {
+    this.logView.addNewRow(object);
+    this.updateFilterCheckboxes(object.log);
+  }
+
+  updateFilterCheckboxes(log) {
+    for (const key of Object.keys(log)) {
+      switch (key) {
+        case 'id':
+          this.addNewCheckbox(this.extCheckboxList, log.id);
+          break;
+        case 'viewType':
+          this.addNewCheckbox(this.viewTypeCheckboxList, log.viewType);
+          break;
+        case 'type':
+          this.addNewCheckbox(this.apiTypeCheckboxList, log.type);
+          break;
+        default:
+          break;
+      }
+    }
   }
 
-  addTableRows(logs) {
-    this.logTable.addNewRows(logs);
+  addNewCheckbox(checkboxList, labelText) {
+    if (!this.isCheckboxExist(checkboxList, labelText)) {
+      const newCheckbox = this.checkboxTemplate.cloneNode(true);
+      newCheckbox.querySelector('label p').textContent = labelText;
+      newCheckbox.querySelector('input').value = labelText;
+      newCheckbox.querySelector('input').checked = true;
+      checkboxList.appendChild(newCheckbox);
+    }
+  }
+
+  isCheckboxExist(checkboxList, labelText) {
+    const checkboxLabels = checkboxList.querySelectorAll('label p');
+    for (const label of checkboxLabels) {
+      if (label.textContent === labelText) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  showRows({ filterParam, filterObject, isFilterMatchingFn }) {
```

Comment:
`showRows` and `hideRows` are almost identical.

`hideRows` is currently only slightly different (i.e. less parameters and less code) because you have hard-coded the filter-type (identity, `===`). When you're going to implement different kinds of filters, such as substring-based, then `hideRows` would also take similar arguments.

To simplify the code, and to remove the needlessly hardcoded assumption on the filter type, I suggest to just loop over all rows, and update the `hidden` state depending on whether the filter matches.

The current two methods select a different set of elements (`tr[hidden]` vs `tr:not([hidden])`) instead of all rows. This is a slighy micro-optimization, which only works because a filter state change (i.e. (un)checking a checkbox) can only affect a strict subset of all rows. When substring-based filtering is implemented, that assumption cannot be relied upon, because a state change (e.g. substring change) can potentially affect all rows. For this reason, I recommend again to apply the filter to all rows at first. We can optimize later, when the functionality itself works.

Reply:
> To simplify the code, and to remove the needlessly hardcoded assumption on the filter type, I suggest to just loop over all rows, and update the `hidden` state depending on whether the filter matches.

Done this :-)